File: PWKeyboardDemo/PWKeyboard/UIView+PWCorners.m
//
//  UIView+PWCorners.m
//  HybridPlateKeyboard
//
//  Created by fzy on 2017/11/3.
//  Copyright © 2017年 fzy. All rights reserved.
//

#import "UIView+PWCorners.h"

@implementation UIView (PWCorners)

- (void)addRoundedCorners:(UIRectCorner)corners
                withRadii:(CGSize)radii {
    
    UIBezierPath* rounded = [UIBezierPath bezierPathWithRoundedRect:self.bounds byRoundingCorners:corners cornerRadii:radii];
    CAShapeLayer* shape = [[CAShapeLayer alloc] init];
    [shape setPath:rounded.CGPath];
    
    self.layer.mask = shape;
}

@end

File: Source/UIView+PWCorners.swift
//
//  UIView+PWCorners.swift
//  VehicleKeyboardDemo
//
//  Created by 杨志豪 on 2018/7/6.
//  Copyright © 2018年 yangzhihao. All rights reserved.
//

import Foundation
import UIKit

extension UIView {
    func addRounded(cornevrs:UIRectCorner, radii:CGSize){
        let rounded = UIBezierPath(roundedRect: bounds, byRoundingCorners: cornevrs, cornerRadii: radii)
        let shape = CAShapeLayer()
        shape.path = rounded.cgPath
        self.layer.mask = shape;
    }
}
